### DISCUSSION TITLE
Getting error, fatal : Could not read from remote repository.  Please make sure you have the correct access rights and the repository exists."

### DISCUSSION THREAD

**Discussion by Praveenab131** [community/community#27289] [New to GitHub]
kex_exchange_identification: read: Connection reset by peer. 
fatal : Could not read from remote repository.  Please make sure you have the correct access rights and the repository exists.

I am getting this issue on trying to push my code for the first time from local to github repository. Please help me to eradicate this issue.

I followed all the ssh steps. Generate, Add, Eval, profile settings(add ssh) in github. Then init git, add, commit, add remote origin and push. push is not working.

**Comment by airtower-luna** [comment]
Please post the *full* command and error message (you can censor private repository names etc.), otherwise it's impossible to answer.

In general, you can use `ssh -vT git@github.com` to debug the SSH layer.

**Reply by RookieRascal** [reply]
thank you so much bro

**Reply by pampa89d** [reply]
I had the same error. In my case the problem was that I used "sudo git clone" instead "git clone".

**Reply by grinapo** [reply]
And for me that one of my ssh keys was used in a test account of mine, and github "guess" the username from key. :frowning_face:

**Reply by glimmbo** [reply]
My context: my frontend dev container was not auth'd, but my backend one was. This command itself solved the problem for me. 👍

**Comment by kurnakovv** [comment]
@Praveenab131 maybe you should create a new repository on github? Because there are no repositories in your profile
after that use
`
git remote add origin git@github.Praveenab131/new_repo
`

`
git push -u origin main
`

**Comment by Praveenab131** [comment]
I have my project only in organization’s repository and is in Private. The repository exists but not under my name

From: KurnakovMaksim ***@***.***>
Sent: 27 July 2022 17:12
To: community/community ***@***.***>
Cc: Praveena B131 ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [community/community] Getting error, fatal : Could not read from remote repository. Please make sure you have the correct access rights and the repository exists." (Discussion #27289)

@Praveenab131 maybe you should create a new repository on github? Because there are no repositories in your profile after that use git remote add origin ***@***.******@***.***/new_repo> git push -u origin master — Reply to this email
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

@Praveenab131<https://github.com/Praveenab131> maybe you should create a new repository on github? Because there are no repositories in your profile
after that use
git remote add origin ***@***.******@***.***/new_repo>

git push -u origin master

**Comment by kurnakovv** [comment]
@Praveenab131 
> The repository exists but not under my name

If this is not your repository, you should make fork, after that clone fork and set remote link on this fork.
Sorry if I understand you not right

**Comment by kurnakovv** [comment]
You can also try to create public repository for test and check ssh work.

**Comment by Praveenab131** [comment]
I troed to Fork but that also giving same issue “Please make sure you have the correct access rights
and the repository exists.”

I should not use public repo.

From: KurnakovMaksim ***@***.***>
Sent: 27 July 2022 17:25
To: community/community ***@***.***>
Cc: Praveena B131 ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [community/community] Getting error, fatal : Could not read from remote repository. Please make sure you have the correct access rights and the repository exists." (Discussion #27289)

You can also try to create public repository for test and check ssh work. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: <community/community/repo-discussions/27289/comments/3260257
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

You can also try to create public repository for test and check ssh work.

**Comment by kurnakovv** [comment]
are you sure what you use correct remote reference? This reference must contain your name and fork repository name:
`
git remote set-url origin git@github.com:Praveenab131/your_fork_repository_name.git
`

**Comment by Praveenab131** [comment]
***@***.******@***.***:Praveenab131/TSB_IBM_WebAutomation_Praveena.git>

From: KurnakovMaksim ***@***.***>
Sent: 27 July 2022 17:45
To: community/community ***@***.***>
Cc: Praveena B131 ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [community/community] Getting error, fatal : Could not read from remote repository. Please make sure you have the correct access rights and the repository exists." (Discussion #27289)

are you sure what you use correct remote reference? This reference must contain your name and fork repository name: git remote set-url origin ***@***.******@***.***:Praveenab131/your_fork_repository_name.git> — Reply to this email directly, ‍
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

are you sure what you use correct remote reference? This reference must contain your name and fork repository name:
git remote set-url origin ***@***.******@***.***:Praveenab131/your_fork_repository_name.git>

**Comment by kurnakovv** [comment]
Can you share git clone command? You can edit real names

**Comment by Praveenab131** [comment]
git clone https://github.com/TSB-Bank/TSB_IBM_WebAutomation.git -  private repo...
after fork:
git clone ***@***.******@***.***:Praveenab131/TSB_IBM_WebAutomation_Praveena.git> - private repo


From: KurnakovMaksim ***@***.***>
Sent: 27 July 2022 17:50
To: community/community ***@***.***>
Cc: Praveena B131 ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [community/community] Getting error, fatal : Could not read from remote repository. Please make sure you have the correct access rights and the repository exists." (Discussion #27289)

Can you share git clone command? You can edit real names — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: <community/community/repo-discussions/27289/comments/3260434
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

Can you share git clone command? You can edit real names

**Comment by kurnakovv** [comment]
> TSB_IBM_WebAutomation_Praveena

Remove _Praveena

`git clone git@github.com:Praveenab131/TSB_IBM_WebAutomation.git`

**Comment by Praveenab131** [comment]
That is my repo name. Nothing wrong with that


From: KurnakovMaksim ***@***.***>
Sent: 27 July 2022 18:40
To: community/community ***@***.***>
Cc: Praveena B131 ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [community/community] Getting error, fatal : Could not read from remote repository. Please make sure you have the correct access rights and the repository exists." (Discussion #27289)

TSB_IBM_WebAutomation_Praveena Remove _Praveena — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you were mentioned.Message ID: <community/community/repo-discussions/27289/comments/3260771
ZjQcmQRYFpfptBannerStart
This Message Is From an External Sender
This message came from outside your organization.
ZjQcmQRYFpfptBannerEnd

TSB_IBM_WebAutomation_Praveena

Remove _Praveena

**Comment by kurnakovv** [comment]
what console return if you use `git clone git@github.com:Praveenab131/TSB_IBM_WebAutomation.git`?

**Comment by ghost** [comment]
Hi there,

I'm trying to push code to my remote repo but facing an error 
"ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists."

I have previously change my OS window after I have generate my ssh key but now when I do `git push -u` I get an error.

Help me

**Reply by louiewh** [reply]
I also happen to this error

**Reply by mhromyk** [reply]
here is what is left for you.
You need to create repository on github via github web UI. 
I had the same issue till the time I created it via web page then do what you did before. It should work.

**Comment by SharipovHuseyn** [comment]
I have the same problem too (

**Comment by akashdip2001** [comment]
The error you're encountering, `kex_exchange_identification: read: Connection reset by peer`, often indicates a network or SSH configuration issue. Here are steps to troubleshoot and resolve the problem:

1. **Check Network Connection:**
   - Ensure that your internet connection is stable.
   - Try other network activities to confirm that your connection is not the issue.

2. **Verify SSH Key:**
   - Double-check that you've added the correct SSH key to your GitHub account.
   - Confirm that the SSH key in your local SSH agent matches the one associated with your GitHub account.

3. **Check GitHub Access:**
   - Verify that your GitHub account has the necessary permissions to push to the repository.
   - Make sure the repository exists and you are using the correct URL.

4. **Use SSH URL:**
   - Ensure that you are using the SSH URL when adding the remote origin:
     ```
     git remote add origin git@github.com:username/repo.git
     ```

5. **SSH Config:**
   - Confirm that your SSH configuration file (`~/.ssh/config`) doesn't have conflicting settings.
   - If you don't have a config file, this might not be the issue.

6. **Update Git:**
   - Ensure you are using the latest version of Git.
   - Update Git to the latest version if needed.

7. **Firewall/Antivirus:**
   - Check if your firewall or antivirus software is blocking the connection.
   - Temporarily disable firewall/antivirus and attempt the push.

8. **Use HTTPS URL:**
   - As a temporary workaround, you can try using the HTTPS URL for the remote origin.
     ```
     git remote set-url origin https://github.com/username/repo.git
     ```
   - This may not be a permanent solution, but it helps identify if the issue is related to SSH.

9. **Diagnostic Commands:**
   - Run the following commands to gather more information:
     ```
     ssh -vT git@github.com
     ```

After performing these steps, you should have a clearer understanding of the issue. If the problem persists, review the output of the diagnostic commands for more specific error messages and consider reaching out to GitHub support for assistance.

## Profile
![Profile Image](https://thumbnails.odycdn.com/optimize/s:200:0/quality:95/plain/https://thumbs.odycdn.com/bf58cf4d7ec52c62f02c52e16f150a5b.webp)
**AKASHDIP MAHAPATRA**
"everything is physics bro"  
E=mc²

|[Subscribe on YouTube](https://www.youtube.com/channel/UCxvmp634YDc41xCWOdvWqoQ?sub_confirmation=1) |
|----  |

**Reply by NellyJazminPC** [reply]
I have been trying everything and it worked until I disconnected my VPN, I forgot I had it on. I disconnected it and tried "push" and all fine. 🙌

**Reply by MarinaSergeyeva** [reply]
You saved me with this comment! Thank you from all my heart 😂

**Comment by tarun9689-sharma** [comment]
Hi I'm getting the below error

ERROR: Write access to repository not granted.
fatal: Could not read from remote repository.


i've the access and required things but suddenly getting this

**Comment by Nuxview** [comment]
Hey, I'm also getting the same error yet I confirmed that the repo is created and public.

ERROR: Repository not found.
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

What I don't understand is that I pushed my local repo, yesterday as of writing, via SSH without any problems.

**Reply by GeorgeLMaluf0579** [reply]
Hi. I face the same error. But I discover a way to work.
In my case was because the ssh key files was not named like the standard.
I create a file name _config_ inside on my local .ssh directory.
Here is the basic contents:
```
HostName github.com
   HostName github.com
   IdentityFile _filename_of_private_filekey_ 
   IdentitiesOnly true
```

**Reply by Nuxview** [reply]
Thanks for the reply but I fixed my error. I found out that the generated SSH key is different for both the super user and the local user and the repos I was trying to create had a mix of non-sudo and sudo commands since they were for NixOS configurations which are only accessible if your a super user.

The commands for its setup aren't meant to be run as sudo but being they were configuration files, I had to run them as sudo.

Long story short, each user has a different SSH key so make sure to create and add both keys.
